test/Dockerfile.test-distroless

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### Dockerfile for building a distroless image for testing
+# Modified version of https://github.com/kubernetes/release/blob/master/images/build/distroless-iptables/distroless/Dockerfile
+
+ARG STAGE_DIR="/opt/stage"
+
+FROM debian:bullseye-slim as build
+ARG STAGE_DIR
+
+COPY test/build/stage-binaries-from-package.sh /
+COPY test/build/package-utils.sh /
+COPY test/build/stage-binary-and-deps.sh /
+
+RUN mkdir -p "${STAGE_DIR}" && \
+    /stage-binaries-from-package.sh "${STAGE_DIR}" conntrack \
+    ebtables    \
+    ipset       \
+    iptables    \
+    kmod		&& \
+	`# below binaries and dash are used by iptables-wrapper-installer.sh` \
+    /stage-binary-and-deps.sh "${STAGE_DIR}" /bin/dash \
+    /bin/mv \
+    /bin/chmod \
+    /bin/grep \
+    /bin/ln  \
+    /bin/rm \
+    /bin/sleep \
+    /usr/bin/wc
+
+RUN ln -sf /bin/dash "${STAGE_DIR}"/bin/sh
+
+FROM gcr.io/distroless/static as intermediate
+ARG STAGE_DIR
+
+COPY test/build/clean-distroless.sh /clean-distroless.sh
+COPY --from=build "${STAGE_DIR}" /
+COPY iptables-wrapper-installer.sh /
+COPY bin/iptables-wrapper /
+# iptables-wrapper-installer needs to know that iptables exists before doing all its magic
+RUN echo "" >  /usr/sbin/iptables && \
+	/iptables-wrapper-installer.sh && \
+    /clean-distroless.sh
+
+FROM scratch
+
+COPY --from=intermediate / /
```

Comment:
If you copy everything from `intermediate` into the final image, then why not just use the "intermediate" image as the final result?

Anyway, I think `Dockerfile.test-distroless` should look something like:

```
FROM debian:bullseye-slim as build

# FIXME install iptables
# FIXME use dpkg to figure out all the packages that iptables depends on
# FIXME use dpkg to figure out all of the files in all of the packages from the previous list
# FIXME copy all of those files to ${STAGE_DIR}

FROM gcr.io/distroless/static
COPY --from=build "${STAGE_DIR}" /
COPY bin/iptables-wrapper /sbin
RUN /sbin/iptables-wrapper --install

COPY bin/tests /
```

Reply:
Yeah this was great feedback. This is more or less what the dockerfile does now. But as I wrote in a previous comment, I kept the staging scripts.